autoPyTorch/api/base_task.py

```diff
@@ -1237,24 +1384,119 @@ def fit(self,
+                name and Value is an Iterable of the names of the components
+                to include. Only these components will be present in the
+                search space.
+            exclude_components (Optional[Dict[str, Any]]):
+                Dictionary containing components to exclude. Key is the node
+                name and Value is an Iterable of the names of the components
+                to exclude. All except these components will be present in
+                the search space.
+            search_space_updates(Optional[HyperparameterSearchSpaceUpdates]):
+                Updates to be made to the hyperparameter search space of the pipeline
+            budget (Optional[float]):
+                Budget to fit a single run of the pipeline. If not
+                provided, uses the default in the pipeline config
+            pipeline_options (Optional[Dict]):
+                Valid config options include "device",
+                "torch_num_threads", "early_stopping", "use_tensorboard_logger",
+                "metrics_during_training"
+            disable_file_output (Optional[List[Union[str, DisableFileOutputParameters]]]):
+                Used as a list to pass more fine-grained
+                information on what to save. Must be a member of `DisableFileOutputParameters`.
+                Allowed elements in the list are:
+
+                + `y_optimization`:
+                    do not save the predictions for the optimization set,
+                    which would later on be used to build an ensemble. Note that SMAC
+                    optimizes a metric evaluated on the optimization set.
+                + `pipeline`:
+                    do not save any individual pipeline files
+                + `pipelines`:
+                    In case of cross validation, disables saving the joint model of the
+                    pipelines fit on each fold.
+                + `y_test`:
+                    do not save the predictions for the test set.
+                + `all`:
+                    do not save any of the above.
+                For more information check `autoPyTorch.evaluation.utils.DisableFileOutputParameters`.
+            configuration: (Configuration)
+                configuration to fit the pipeline with.
 
         Returns:
-            BasePipeline:
+            (BasePipeline):
                 fitted pipeline
+            (RunInfo):
+                Run information
+            (RunValue):
+                Result of fitting the pipeline
+            (BaseDataset):
+                Dataset created from the given tensors
         """
-        self.dataset_name = dataset.dataset_name
 
-        if self._logger is None:
-            self._logger = self._get_logger(str(self.dataset_name))
+        if dataset is None:
+            assert X_train is not None and \
+                   y_train is not None, "No dataset provided, must provide X_train, y_train tensors"
```

Comment:
```suggestion
            if X_train is None or y_train is None:
                raise ValueError("X_train and y_train must be provided")
```

Reply:
Actually, `get_dataset` expects there to be `X_train` and `y_train`, python itself will raise an error. I have this check because we allow the user to either send a dataset object or `X_train`, etc. 

I have done so because in case the user wants to get the default configuration or sample a configuration like [here](https://github.com/ravinkohli/Auto-PyTorch/blob/a0fef7700fd0be8696db1c8a477d4ddedf4f6754/examples/40_advanced/example_single_configuration.py#L59), they will have to create their own dataset object and hence they can use it to fit a pipeline. I have also allowed the option for `X_train`, etc. in case the user would like to fit an incumbent configuration after a search, they won't need to instantiate the dataset object themselves.

Reply:
Yeah, if it is literally `get_dataset`, it is better to stick to your implementation, thanks for the explanation.
But could you change assert to raise error?

Reply:
sure.